cli/cmd/install.go

```diff
@@ -280,59 +433,77 @@ func readIntoBytes(filename string) ([]byte, error) {
 	return buf.Bytes(), nil
 }
 
-func globalConfig(options *installOptions) *config.Global {
-	var identityContext *config.IdentityContext
+func globalConfig(options *installOptions, id *installIdentityConfig) *pb.Global {
+	var identityContext *pb.IdentityContext
+	if id != nil {
+		il, err := time.ParseDuration(id.Issuer.IssuanceLifetime)
+		if err != nil {
+			il = defaultIdentityIssuanceLifetime
+		}
+
+		csa, err := time.ParseDuration(id.Issuer.ClockSkewAllowance)
+		if err != nil {
+			csa = defaultIdentityClockSkewAllowance
+		}
+
+		identityContext = &pb.IdentityContext{
+			TrustDomain:        id.TrustDomain,
+			TrustAnchorsPem:    id.TrustAnchorsPEM,
+			IssuanceLifetime:   ptypes.DurationProto(il),
+			ClockSkewAllowance: ptypes.DurationProto(csa),
+		}
```

Comment:
The code inside this if condition is identical to the code that constructs the `IdentityContext` on lines 388-406 above. Consider moving it to a shared func. Something like:

```diff
diff --git a/cli/cmd/install.go b/cli/cmd/install.go
index 52d8b328..07ce852f 100644
--- a/cli/cmd/install.go
+++ b/cli/cmd/install.go
@@ -384,25 +384,7 @@ func render(config installConfig, w io.Writer, options *installOptions) error {
 	// injectOptionsToConfigs does NOT set an identity context if none exists,
 	// since it can't be enabled at inject-time if it's not enabled at
 	// install-time.
-	if config.Identity != nil {
-		id := config.Identity
-		il, err := time.ParseDuration(id.Issuer.IssuanceLifetime)
-		if err != nil {
-			il = defaultIdentityIssuanceLifetime
-		}
-
-		csa, err := time.ParseDuration(id.Issuer.ClockSkewAllowance)
-		if err != nil {
-			csa = defaultIdentityClockSkewAllowance
-		}
-
-		pbConfig.global.IdentityContext = &pb.IdentityContext{
-			TrustDomain:        id.TrustDomain,
-			TrustAnchorsPem:    id.TrustAnchorsPEM,
-			IssuanceLifetime:   ptypes.DurationProto(il),
-			ClockSkewAllowance: ptypes.DurationProto(csa),
-		}
-	}
+	pbConfig.global.IdentityContext = config.Identity.toIdentityContext()
 
 	return processYAML(&buf, w, ioutil.Discard, resourceTransformerInject{
 		configs: pbConfig,
@@ -434,31 +416,11 @@ func readIntoBytes(filename string) ([]byte, error) {
 }
 
 func globalConfig(options *installOptions, id *installIdentityConfig) *pb.Global {
-	var identityContext *pb.IdentityContext
-	if id != nil {
-		il, err := time.ParseDuration(id.Issuer.IssuanceLifetime)
-		if err != nil {
-			il = defaultIdentityIssuanceLifetime
-		}
-
-		csa, err := time.ParseDuration(id.Issuer.ClockSkewAllowance)
-		if err != nil {
-			csa = defaultIdentityClockSkewAllowance
-		}
-
-		identityContext = &pb.IdentityContext{
-			TrustDomain:        id.TrustDomain,
-			TrustAnchorsPem:    id.TrustAnchorsPEM,
-			IssuanceLifetime:   ptypes.DurationProto(il),
-			ClockSkewAllowance: ptypes.DurationProto(csa),
-		}
-	}
-
 	return &pb.Global{
 		LinkerdNamespace: controlPlaneNamespace,
 		CniEnabled:       options.noInitContainer,
 		Version:          options.linkerdVersion,
-		IdentityContext:  identityContext,
+		IdentityContext:  id.toIdentityContext(),
 	}
 }
 
@@ -509,3 +471,26 @@ func proxyConfig(options *installOptions) *pb.Proxy {
 		DisableExternalProfiles: options.disableExternalProfiles,
 	}
 }
+
+func (id *installIdentityConfig) toIdentityContext() *pb.IdentityContext {
+	if id == nil {
+		return nil
+	}
+
+	il, err := time.ParseDuration(id.Issuer.IssuanceLifetime)
+	if err != nil {
+		il = defaultIdentityIssuanceLifetime
+	}
+
+	csa, err := time.ParseDuration(id.Issuer.ClockSkewAllowance)
+	if err != nil {
+		csa = defaultIdentityClockSkewAllowance
+	}
+
+	return &pb.IdentityContext{
+		TrustDomain:        id.TrustDomain,
+		TrustAnchorsPem:    id.TrustAnchorsPEM,
+		IssuanceLifetime:   ptypes.DurationProto(il),
+		ClockSkewAllowance: ptypes.DurationProto(csa),
+	}
+}
```